Develop3

BlackSunVRN opened the pull request:
Помимо основных требований к заданий был использован rest assured для удаления пользователя после создания в тестах.

BlackSunVRN committed:
pom.xml edited
added tests for Login, Register, Logout and MainPage
added model and clients for delete user after test
added pageObjects for all pages with needed tests

BlackSunVRN committed:
add allure report

pomegranate282922 commented on src/main/java/pageObject/LoginPage.java:
```diff
@@ -0,0 +1,68 @@
+package pageObject;
```
Пакеты именуем без camelCase

BlackSunVRN replied:
Исправлено.

pomegranate282922 commented on src/test/java/LoginTest.java:
```diff
@@ -0,0 +1,89 @@
+
+public class LoginTest {
+
+    public static final String TEST_EMAIL = "testBoldin@ya.ru";
```
Нет, так нехорошо. Твой тест зависит от каких-то данных в бд. Надо создать рандомизированного юзера, с ним работать в тесте, в конце удалить.

BlackSunVRN replied:
Добавил создание рандомного пользователя.

pomegranate282922 commented on src/test/java/LogoutTest.java:
```diff
@@ -0,0 +1,41 @@
+    @Test
+    @DisplayName("Проверка выхода из профиля после авторизации")
+    public void shouldBeLogout() {
+        loginPage = mainPage.clickLoginButton();
```
Здесь тоже создать юзера надо. Через апи.

BlackSunVRN replied:
Добавил создание рандомного пользователя.

pomegranate282922 commented on src/test/java/RegisterTest.java:
```diff
@@ -0,0 +1,71 @@
+        registerPage = loginPage.clickRegister();
+        registerPage.setPassword(RandomStringUtils.randomAlphabetic(5));
+        registerPage.clickRegisterButton();
+        registerPage.compareText("Некорректный пароль");
```
А если баг? Тоже нужно пытаться удалить юзера!

BlackSunVRN replied:
Тут не заполняются два обязательных поля для создания пользователя, т.е. в случае, даже если пароль прошел, тест все равно упадет, т.к. юзер не создан без двух других полей.
Добавил заполнение двух других полей, чтобы в случае, если возник баг и пароль из 5 символов подошел, пользователь был все равно создан. Далее в @After добавил try на удаление юзера, а в случае, если пользователя нет, пишем сообщение, что так и должно быть, т.к. он и не был создан (т.е. тест пройдет успешно).

pomegranate282922 reviewed (commented):
Здесь надо немного доделать - создавай рандомизированного юзера через апи, его используй в тестах, удаляй в конце!
В тесте на создание юзера предусмотри удаление ошибочно созданного юзера!

BlackSunVRN committed:
remove camelCase from package name
add random user generate for login and register tests
add try catch for delete user after register

BlackSunVRN committed:
add lombok with annotations
fix OpenProfileTest and OpenConstructorTest for use generated with api users
add static variables for urls and its parts

pomegranate282922 reviewed (approved):
Зачёт!
